subgraphs/_reference_/src/sdk/protocols/generic/index.ts

```diff
@@ -0,0 +1,147 @@
+import { Address, Bytes, BigInt, ethereum } from "@graphprotocol/graph-ts";
+
+import { ProtocolManager } from "./protocol";
+import { EventsManager } from "./event";
+import { PoolsManager } from "./pool";
+import { AccountsManager } from "./account";
+import { TokensManager, TokenInitializer } from "./token";
+import { TokenPricer } from "../config";
+import { Config } from "./config";
+import { BIGINT_ZERO, ZERO_ADDRESS } from "../../util/constants";
+
+export class CustomEventType {
+  block: ethereum.Block;
+  transaction: ethereum.Transaction;
+  logIndex: BigInt;
+  event: ethereum.Event | null;
+
+  constructor() {
+    this.block = new ethereum.Block(
+      Bytes.empty(),
+      Bytes.empty(),
+      Bytes.empty(),
+      Address.fromString(ZERO_ADDRESS),
+      Bytes.empty(),
+      Bytes.empty(),
+      Bytes.empty(),
+      BIGINT_ZERO,
+      BIGINT_ZERO,
+      BIGINT_ZERO,
+      BIGINT_ZERO,
+      BIGINT_ZERO,
+      BIGINT_ZERO,
+      null,
+      null
+    );
+    this.transaction = new ethereum.Transaction(
+      Bytes.empty(),
+      BIGINT_ZERO,
+      Address.fromString(ZERO_ADDRESS),
+      null,
+      BIGINT_ZERO,
+      BIGINT_ZERO,
+      BIGINT_ZERO,
+      Bytes.empty(),
+      BIGINT_ZERO
+    );
+    this.logIndex = BIGINT_ZERO;
+    this.event = null;
+  }
+
+  static initialize(
```

Comment:
I know you copied this as it was from bridges, but since we are here …

To reduce verbosity or redundant arguments, you can add 2 other methods like `initializeFromEvent` and `initializeFromCall` (or without the “initialize” part), to make it straightforward and simpler to call.

Alternatively, instead of requiring the SDK to pass a CustomEvent to the constructor, you can do the same thing there, having a couple static methods that serve as constructors: `SDK.initFromEvent()` or `SDK.initFromCall`. I prefer to be explicit over having an `initialize<T>` with a generic argument.

Reply:
Okay - I added separate static methods.